### PR TITLE
Fix enum and flag repr in struct reprs

### LINKED ISSUE #100: cstruct v4 prints enums as integers instead of the enum itself
In `dissect.cstruct` v4 the enum types are now printed as hex integers instead of the enum type itself.
This affects printing a structure but also in dumpstruct. In my opinion printing only the integer loses some contextual information that is useful when dumping structures.

The `examples/mirai.py` in the cstruct repository shows the difference in after and before:

dissect.cstruct==4.1:

image

dissect.cstruct==3.9:

image

Notice the difference in `AttackOption type=`

### PR DESCRIPTION
**Pull request by Schamper**
Fixes #100

### CODE REVIEW

File: dissect/cstruct/types/structure.py [hunk 1]
```diff
@@ -380,10 +381,15 @@ def __getitem__(self, item: str) -> Any:
         return getattr(self, item)
 
     def __repr__(self) -> str:
-        values = [
-            f"{k}={hex(self[k]) if (issubclass(f.type, int) and not issubclass(f.type, Pointer)) else repr(self[k])}"
-            for k, f in self.__class__.fields.items()
-        ]
+        values = []
+        for k, f in self.__class__.fields.items():
```
Comment (Miauwkeru) [inline dissect/cstruct/types/structure.py]: no single character variable names pls

**Review by Miauwkeru — DISMISSED** [review]

**Comment by yunzheng** [comment]
please also fix the same behavior in `dumpstruct()` https://github.com/fox-it/dissect.cstruct/blob/046dfa51f0989c7f55cf6d7dd00ba71f3ec5a115/dissect/cstruct/utils.py#L163

Example output from a `dissect.cobaltstrike` test:
image

**Review by yunzheng — CHANGES_REQUESTED** [review]
please also fix the behaviour in `dumpstruct()`

**Comment by Schamper** [comment]
@yunzheng should be fixed now too.

**Review by yunzheng — APPROVED** [review]
Confirmed that this is now fixed in both repr and dumpstruct output.

**Pull request merged**